### DISCUSSION TITLE
web3 auth logs out

### DISCUSSION THREAD

**Discussion by harshshaw** [Web3Auth/Web3Auth#1206] [Help]
On our Xade app everytime user tries to login using web3auth and then on refreshing the user gets automatically loggedout, could you help us with this issue? 

Link :-https://shardeum.app.xade.finance/

**Comment by yashovardhan** [comment]
Please share your web3auth specific code snippets for us to debug.

**Comment by harshalmadnani** [comment]
The snippets for our web3auth code can be found here 
https://github.com/XadeFinance/webapp/blob/shardeum/src/services/web3auth.tsx

**Comment by MarcosGin** [comment]
Hello, I was having this same error.

Update to the latest versions of the libraries that I am using, from what I saw, you are using very old versions. You may have to refactor your code for these new versions to work.
```
"@web3auth/phantom-adapter": "^4.2.2",
     "@web3auth/torus-solana-adapter": "^4.2.2",
     "@web3auth/base": "^4.2.2",
     "@web3auth/core": "^4.2.2",
     "@web3auth/openlogin-adapter": "^4.2.2",
     "@web3auth/solana-provider": "^4.2.2",
```
